contrib/win32/win32compat/fileio.c

```diff
@@ -308,16 +340,58 @@ createFile_flags_setup(int flags, int mode, struct createFile_flags* cf_flags)
 
 	cf_flags->dwFlagsAndAttributes = FILE_FLAG_OVERLAPPED | SECURITY_IMPERSONATION | FILE_FLAG_BACKUP_SEMANTICS;
 
-	/*TODO - map mode */
+	/*map mode*/
+	if ((pwd = getpwuid(0)) == NULL)
+		fatal("getpwuid failed.");
 
-	return 0;
+	if ((sid_utf16 = utf8_to_utf16(pwd->pw_sid)) == NULL) {
+		debug3("Failed to get utf16 of the sid string");
+		errno = ENOMEM;
+		goto cleanup;
+	}
+	if ((mode & S_IRWXU) != 0) {
+		if (st_mode_to_file_att((mode & S_IRWXU) >> 6, owner_access, sizeof(owner_access)/sizeof(wchar_t)) != 0) {		
+			debug3("st_mode_to_file_att()");
+			goto cleanup;
+		}
+		swprintf(owner_ace, sizeof(owner_ace)/sizeof(wchar_t) - 1, L"(A;;%s;;;%s)", owner_access, sid_utf16);
```

Comment:
Use the secure version of swprintf (https://msdn.microsoft.com/en-us/library/ybk95axf.aspx).